### PR TITLE
Fixes to allow building with GCC and Clang

### PR DESCRIPTION
**Pull request by bsmiles32**
This should fix https://github.com/mupen64plus/mupen64plus-user-issues/issues/689
Also fixes various warnings.

### CODE REVIEW

File: src/Util.cpp [hunk 1]
```diff
@@ -1247,7 +1247,7 @@ void clip_tri (WORD linew)
                 if (settings.ucode == 5)
                     for (i=0; i<n; i++)
                         if (rdp.vtxbuf[i].z < -1000.0f)  return;//rdp.vtxbuf[i].z = 0.0f;
-                        DepthBuffer(rdp.vtx_buffer?(vtx_list2):(vtx_list1), n);
+                DepthBuffer(rdp.vtx_buffer?(vtx_list2):(vtx_list1), n);
```
Comment (Narann) [inline src/Util.cpp]: Wow! This one is weird. The original indents looks like an error.

But, yeah, you don't change any behavior so it's ok...

**Comment by charlemagnelasse** [comment]
@bsmiles32 @richard42 

Please remove the min/max define in src/TexModCI.h and fix the remaining build. Because right now it doesn't build under _WIN32 anymore due to these platform specific defines.

And there are still min/max defines in:

* src/rdp.h
* src/TexModCI.h
* src/TexBuffer.cpp
* src/Util.cpp
* src/ucode06.h

**Comment by bsmiles32** [comment]
@charlemagnelasse Thanks for catching this and the detailed way of fixing it.
Can you test if PR https://github.com/mupen64plus/mupen64plus-video-glide64/pull/14 fixes win32 builds ?

**Comment by charlemagnelasse** [comment]
Sorry, I was away for some days and didn't read the other mails. But richards42 fix seems to work fine

**Pull request merged**